### ISSUE TITLE
Invalid URL

### ISSUE THREAD

**Issue by flammablehavocx** [izqalan/cy-client#61]
cy client v.2.2.0.0 Windows 7 Professional  7601 SP1

Unable to download any album. Tried both x86 and x64 versions,

![1](https://user-images.githubusercontent.com/91030000/133940644-bf856e84-18d6-4da7-bbe4-49e4c63d7881.jpg)

**Comment by Anequit** [comment]
Is your album public?

**Comment by Anequit** [comment]
I just copied and pasted the url into the downloader and it's working flawlessly for me

**Comment by Anequit** [comment]
It could be because you're on windows 7 however. I don't know if the downloader supports such an old OS.

**Comment by Anequit** [comment]
![image](https://user-images.githubusercontent.com/51097472/136711998-b053de14-731d-4c05-96af-9ed0ef2294d2.png)

It's working for me. Download the latest version again just incase something went wrong.

**Comment by Bastek** [comment]
I go exactly same issue at Win 10. I have downloaded latest version cy client several times and every time i get "invalid url" error.

**Comment by Anequit** [comment]
> I got exactly same issue at Win 10. I have downloaded latest version cy client several times and every time i get "invalid url" error.

Clear browser cache and try to reopen the album on your web browser. If it was deleted, then the downloader will report "invalid url". Cyberdrop does have an issue with albums still opening after being deleted due to cache. I will still look into it and see if it's something else.

**Comment by Anequit** [comment]
One other thing is that you need to make sure you are using a .me url and not a .to url. The downloader only downloads from the .me not the .to, but if it's a .me and still saying invalid then it might have been deleted and in that case I can't do anything about that.

**Comment by Bastek** [comment]
Cache cleared, album is active and domain is .me - still invalid url. In fact i tested on 16 albums and result is always the same.

**Comment by Anequit** [comment]
> Cache cleared, album is active and domain is .me - still invalid url. In fact i tested on 16 albums and result is always the same.

That's really odd... I'm having no issues myself and tested with two different computers. Must be something to do with your OS or environment. I honestly can't think of anything that could cause that.

**Comment by Anequit** [comment]
@Bastek I will be releasing a hotfix for this issue. Let me know if it fixes your problem.

**Comment by Bastek** [comment]
@Anequit Sure, could you make .exe with hotfix included?

**Comment by Anequit** [comment]
> @Anequit Sure, could you make .exe with hotfix included?

I'll release hotfix as update so you'll be able to download the latest version from the releases page

**Comment by Bastek** [comment]
Sadly, the issue is still there. Win 10 pro 10.0.19043, Microsoft .NET Runtime 5.0.12 (x64). I tested on album mentioned in this discussion: https://cyberdrop.me/a/GfiuT4es

**Comment by Aziz6700** [comment]
> > @Anequit Sure, could you make .exe with hotfix included?
> 
> I'll release hotfix as update so you'll be able to download the latest version from the releases page

still doesn't work man

**Comment by Orligon** [comment]
Have the same problem

**Comment by Bastek** [comment]
@izqalan Yes, all albums are accessible via browser and it loads instantly with no issues at all.

**Comment by Anequit** [comment]
I'm out of town right now but when I get home I'll add some better logging so that we can narrow down what's going wrong

**Comment by SnowVolf** [comment]
Same error on (Windows 10 21H2, Windows 11 22000.384)
![image](https://user-images.githubusercontent.com/14909737/148508913-f9a8673b-6fb6-4d48-a403-da2f5735794d.png)

**Comment by Anequit** [comment]
https://github.com/izqalan/cy-client/releases/tag/v2.2.4

**Comment by Bastek** [comment]
> https://github.com/izqalan/cy-client/releases/tag/v2.2.4

I tried again with 27 download urls and issue is still the same.

**Comment by Anequit** [comment]
What is it telling you now?

**Comment by Bastek** [comment]
Exactly the same: invalid url.

**Comment by Anequit** [comment]
I see... I left in a piece of code that I forgot to remove, so I'll push a new version with that fix.

**Comment by Anequit** [comment]
https://github.com/izqalan/cy-client/releases/tag/v2.2.5

**Comment by Bastek** [comment]
Unknown webscraper error. Please report this to the github repository.

**Comment by Anequit** [comment]
Thanks, this will help narrow down the cause of the issue.

**Comment by Anequit** [comment]
https://github.com/izqalan/cy-client/releases/tag/v.2.2.6

**Comment by Bastek** [comment]
Unknown webscraper error. Please report this to the github repository. Input string was not in a correct format.

I tested on album mentioned in this discussion: https://cyberdrop.me/a/GfiuT4es

**Comment by Anequit** [comment]
I believe I've identified the issue, so I'm going to work on patching it today.

**Comment by Anequit** [comment]
https://github.com/izqalan/cy-client/releases/tag/v2.2.7

**Comment by Bastek** [comment]
![obraz](https://user-images.githubusercontent.com/537738/150702076-825ca3db-a43d-4547-93ff-ff78f573ac35.png)

**Comment by Anequit** [comment]
In a CMD prompt can you type "dotnet --info" and then copy and paste it here?

**Comment by Anequit** [comment]
This issue should now be fixed.

**Comment by Bastek** [comment]
You're right, this issue is now fixed! Thank you for your time and effort!

**Comment by Anequit** [comment]
> You're right, this issue is now fixed! Thank you for your time and effort!

It's my job to find the solution to the issue, thank you for helping me find the issue.
